packages/jest-haste-map/src/index.ts

```diff
@@ -1143,3 +1146,11 @@ function copy<T extends Record<string, unknown>>(object: T): T {
 function copyMap<K, V>(input: Map<K, V>): Map<K, V> {
   return new Map(input);
 }
+
+// Export the smallest API surface required by Jest
+type IJestHasteMap = HasteMapStatic & {
```

Comment:
should this be exported?

Reply:
I'm not sure to be honest. In a way it's exported via `typeof` the default export. Users of `hasteImplModulePath` shouldn't be using it, because they only need to provide a module that implements the smaller `HasteMapStatic` interface (which is exported).

The general idea is to "hide" statics we don't need to be public - such as the bunch that come from extending `EventEmitter`.

Happy to go with your judgement anyway

Reply:
fair enough 👍 If for some reason people want it, I assume they'll send a PR 😀